File: vaarthahub_app/lib/models/magazine.dart
class Magazine {
  final int magazineId;
  final int itemId;
  final int newspaperId;
  final String name;
  final String category;
  final String publicationCycle;
  final double price;
  final String? logoUrl;
  final bool isActive;

  Magazine({
    required this.magazineId,
    required this.itemId,
    required this.newspaperId,
    required this.name,
    required this.category,
    required this.publicationCycle,
    required this.price,
    this.logoUrl,
    required this.isActive,
  });

  factory Magazine.fromJson(Map<String, dynamic> json) {
    return Magazine(
      magazineId: json['magazineId'] ?? 0,
      itemId: json['itemId'] ?? 2,
      newspaperId: json['newspaperId'] ?? 0,
      name: json['name'] ?? '',
      category: json['category'] ?? '',
      publicationCycle: json['publicationCycle'] ?? '',
      price: (json['price'] ?? 0).toDouble(),
      logoUrl: json['logoUrl'],
      isActive: json['isActive'] ?? true,
    );
  }
}

File: vaarthahub_api/vaarthahub_api/Models/Magazine.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace vaarthahub_api.Models
{
    public class Magazine
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int MagazineId { get; set; }

        public int ItemId { get; set; } = 2;

        public string ItemType { get; set; } = "Magazine";

        public int NewspaperId { get; set; }

        public string Name { get; set; } = string.Empty;

        public string Category { get; set; } = string.Empty;

        public string PublicationCycle { get; set; } = string.Empty;

        [Column(TypeName = "decimal(6, 2)")]
        public decimal Price { get; set; }

        public string? LogoUrl { get; set; }

        public bool IsActive { get; set; } = true;

        [ForeignKey("NewspaperId")]
        public virtual Newspaper? Newspaper { get; set; }
    }
}
